### ISSUE TITLE
opendata.dk integration

### ISSUE THREAD

**Issue by JosefAssad** [JosefAssad/SeMaWi#95]

**Comment by JosefAssad** [comment]
@baffioso kan du beskrive kort nogle specifikke use cases?

**Comment by baffioso** [comment]
Open Data DK bruger [CKAN](http://ckan.org/) som @mapcentia er i gang med at lave integration til fra GC2. Vi kommer til at udstille data (og lidt metadata) fra vores GC2 Amazonserver, men det er tanken at data på sigt også skal beriges med de metadata vi samler i Wikien ovre i CKAN. Vi har f.eks. data fra vores belysningsprojekt, som bliver opdateret med dokumentation når #62 kører.

**Comment by JosefAssad** [comment]
Hvis vi kigger på et eksempel på et dataset fra opendata.dk:

side: http://www.portal.opendata.dk/dataset/skraldespande
API representation: http://www.portal.opendata.dk//api/3/action/package_show?id=skraldespande

Jeg gætter, at wikisiden for pågældende geodata tabel skal være i feltet `result.resources[0].description`?

**Comment by JosefAssad** [comment]
Har i nogle tanker omkring, hvordan denne integration skal fungere fra SeMaWi synspunktet?

Jeg går ud fra, at værdierne i vores Template:Geodata som vi jo henter fra GC2 er de primære Mapcentias integration behandler. Er det indholdet i den tilknyttede dokumentationsside der også skal udstilles på opendata.dk (hvad enten via link eller kopi)?

**Comment by baffioso** [comment]
Tænker i første omgang vi starter med link til Wikien. Lars og jeg skal lige have snakket lidt mere om hvad der skal ind i opendata.dk.

**Comment by mapcentia** [comment]
CKAN understøttelse i GC2 er klar til brug.

**Comment by JosefAssad** [comment]
Yeah det er derfor jeg tænker lidt på, hvad der skal indholdsmæssigt ind i CKAN fra SeMaWi i koordination mmed GC2. Jeg forestiller mig det kan være fx. værdien af "Free Text" under Category:Geodata siden.

Vi har teknisk set 3 muligheder:

1. Gc2 og SeMaWi laver deres egne CKAN kald, hver bidrager med det indhold den har
2. GC2 integrerer SeMaWis indhold i sine CKAN kald
3. SeMaWi integrerer GC2s indhold i sine CKAN kald.

Jeg tror option 2 kan være den mest praktiske for denne release. Flere årsager: GC2's CKAN integration er oppe og køre, og fordi vi bruger GC2's GUID'er til geodata dokumentationssider er det faktisk en rimelig nem opgave at hive den dokumentation ud af SeMaWi. Eksempel:

En geodata tabel mmed GUID `$GUID` kan få SeMaWi's dokumentation ved at kalde `http://$HOST/index.php/Geodata_dokumentation/$GUID?action=raw` for at få sidens kilde wikitext, eller bare `http://$HOST/index.php/Geodata_dokumentation/$GUID` hvis der skal linkes (ikke helt overbevist vi bare skal linke, husk serveren bliver intern).

Jeg ved ikke hvilken markup standard CKAN bruger, men pandoc understøtter mediawiki syntax is begge retninger (ikke perfekt men tilstrækkeligt). Så det er heller ikke raketvidenskab at konvertere `?action=raw` til hvad det nu viser sig CKAN bbruger, om det er rst, markdown, eller html.

Hvordan lyder det?

**Comment by JosefAssad** [comment]
@mapcentia tanker?

**Comment by mapcentia** [comment]
GC2 har nu indbygget en CKAN client. Denne klient kunne sagtens trække Wiki teksten fra http://$HOST/index.php/Geodata_dokumentation/$GUID?action=raw og sende med til CKAN. Det vil så være enten i "Beskrivelse" til et datasæt eller et felt(er) under "Yderligere info". Er det formateret tekst er "Beskrivelse" nok bedst - det understøtter markdown.

**Comment by mapcentia** [comment]
Lige en kommentar til mulighed 1: Jeg tror ikke denne er god, da faren for at overskive hinanden er stor. Godt nok benytter GC2 CKANs Patch API, men risikoen er der.

**Comment by JosefAssad** [comment]
Ja jeg synes også option nummer to er den mest praktiske.

Til konvertering fra wikiText til Markdown fungerer `pandoc` tilstrækkelig godt i min erfaring. Dog fører det til wikilinks som ikke bliver medtaget, heller ikke indlejrede billeder.

**Comment by JosefAssad** [comment]
Vi mangler afgørelse her. Flytter til milestone 2016_02.
